src/node.rs

```diff
@@ -708,68 +738,78 @@ impl<S: ForkSource + std::fmt::Debug> InMemoryNode<S> {
     }
 
     /// Runs L2 'eth call' method - that doesn't commit to a block.
-    fn run_l2_call(&self, l2_tx: L2Tx) -> Result<VmBlockResult, String> {
-        let execution_mode = TxExecutionMode::EthCall {
-            missed_storage_invocation_limit: 1000000,
-        };
+    fn run_l2_call(&self, l2_tx: L2Tx) -> Result<ExecutionResult, String> {
+        let execution_mode = TxExecutionMode::EthCall;
 
         let inner = self
             .inner
             .write()
             .map_err(|e| format!("Failed to acquire write lock: {}", e))?;
 
-        let mut storage_view = StorageView::new(&inner.fork_storage);
+        let storage = StorageView::new(&inner.fork_storage).to_rc_ptr();
 
-        let mut oracle_tools = OracleTools::new(&mut storage_view, HistoryEnabled);
+        let bootloader_code = inner.system_contracts.contacts_for_l2_call();
 
-        let bootloader_code = &inner.system_contracts.contacts_for_l2_call();
+        // init vm
 
-        let block_context = inner.create_block_context();
```

Comment:
When you merge from latest `main`, we will also need to make the similar adjustments for `evm_mine` and `hardhat_mine` APIs

Reply:
yep.done.